B 20620 mobile home db model int

KonstanceH opened the pull request:
## [B-20620](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A982507)

## Summary

This story adds the model/db table for a mobile home shipment.
This also renames where it used to be motorhome to match mobilehome.

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### How to test
    Review the code and check for correctness.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

### Database

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots

robot-mymove commented:
<!--
  0 failure: 
  1 warning:  This PR does not ...
  
  
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This PR does not include changes to unit tests, even though it affects app code.</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 893157c9ed953c6e68bc1c8eb00c25067e0e62c3
</p>

JamesHawks224 commented on migrations/app/schema/20240723184614_create_mobile_home.up.sql:
```diff
@@ -0,0 +1,21 @@
+    REFERENCES mto_shipments(id),
+	make varchar NOT NULL,
+	model varchar NOT NULL,
+	mh_year int NOT NULL,
+	mh_length int NOT NULL,
```
Why is year and length prefixed with "mh_".  The other fields are not.  If there's not a compelling reason, they should be made consistent with the other fields.

KonstanceH replied:
I just changed it to mh_year/mh_length because year/length seemed to be sql keywords

KonstanceH replied:
These have been updated to year and length_in_inches

JamesHawks224 commented on migrations/app/schema/20240723184614_create_mobile_home.up.sql:
```diff
@@ -0,0 +1,21 @@
+);
+
+COMMENT on TABLE mobile_home IS 'Stores all mobile home shipments, and their details.';
+COMMENT on COLUMN mobile_home.shipment_id IS 'MTO shipment ID associated with this PPM shipment.';
```
comment is referring to "PPM Shipment", needs to be updated.

JamesHawks224 commented on swagger-def/definitions/MobileHome.yaml:
```diff
@@ -0,0 +1,68 @@
+    format: date-time
+    type: string
+    readOnly: true
+  pickupAddress:
+    $ref: 'Address.yaml'
+  hasSecondaryPickupAddress:
+    type: boolean
+    x-omitempty: false
+    x-nullable: true
+  requestedPickupDate:
+    format: date-time
+    type: string
+  pickupLocation:
+    $ref: 'Address.yaml'
+  requestedDeliveryDate:
+    format: date-time
+    type: string
+  dimensions:
+    description: The complete dimesions of a mobile home. w x l x h?
+    type: string
+  orginAddress:
+    $ref: 'Address.yaml'
+  secondaryPickupAddress:
+    allOf:
+      - $ref: 'Address.yaml'
+      - x-nullable: true
+      - x-omitempty: false
+  receivingAgent:
+    type: string
+  counselorRemarks:
+    type: string
+  customerRemarks:
+    type: string
```
Should these fields be here?  They aren't in the table.

KonstanceH replied:
No I think these fields are tied to the mtoshipment and not unique to the mobile home shipment so I'll remove these.

JamesHawks224 reviewed (changes requested):
Code reviewed.  Comments made.

brooklyn-welsh commented on swagger-def/definitions/MobileHome.yaml:
```diff
@@ -0,0 +1,68 @@
+  dimensions:
+    description: The complete dimesions of a mobile home. w x l x h?
+    type: string
+  orginAddress:
```
Small typo "orgin" -> "origin"

brooklyn-welsh commented on swagger/ghc.yaml:
```diff
@@ -7823,15 +7880,15 @@ definitions:
       - PPM
       - BOAT_HAUL_AWAY
       - BOAT_TOW_AWAY
+      - Mobile_Home
```
I'd probably go ahead and all caps this just to stay consistent.

brooklyn-welsh commented on swagger/ghc.yaml:
```diff
@@ -7823,15 +7880,15 @@ definitions:
       PPM: PPM
-      BOAT_HAUL_AWAY: Boat Haul-Away
-      BOAT_TOW_AWAY: Boat Tow-Away
+      Mobile_Home: Mobile Home
```
Same, "Mobile_Home" -> "MOBILE_HOME"

brooklyn-welsh commented on swagger-def/definitions/MTOShipmentType.yaml:
```diff
@@ -10,12 +10,12 @@ enum:
   - PPM
   - BOAT_HAUL_AWAY
   - BOAT_TOW_AWAY
+  - Mobile_Home
```
All caps for consistency: "Mobile_Home" -> "MOBILE_HOME"

brooklyn-welsh commented on swagger-def/definitions/MTOShipmentType.yaml:
```diff
@@ -10,12 +10,12 @@ enum:
   PPM: PPM
-  BOAT_HAUL_AWAY: Boat Haul-Away
-  BOAT_TOW_AWAY: Boat Tow-Away
+  Mobile_Home: Mobile Home
```
All caps for consistency: "Mobile_Home" -> "MOBILE_HOME"

brooklyn-welsh commented on src/pages/MyMove/Multi-Moves/MultiMovesMoveContainer/MultiMovesMoveContainer.jsx:
```diff
@@ -48,8 +48,8 @@ const MultiMovesMoveContainer = ({ moves, setFlashMessage }) => {
-    if (shipmentType === 'MOTORHOME') {
-      return 'Motorhome';
+    if (shipmentType === 'MOBILEHOME') {
+      return 'Mobilehome';
```
Minor formatting fix: "Mobilehome" -> "Mobile Home"

brooklyn-welsh commented on migrations/app/schema/20200205160112_add_enum_values_to_shipment_type_within_mto_shipments.up.sql:
```diff
@@ -7,7 +7,7 @@ CREATE TYPE mto_shipment_type_2 AS ENUM (
     'HHG_OUTOF_NTS_DOMESTIC',
-    'MOTORHOME',
+    'MOBILEHOME',
     'BOAT_HAUL_AWAY',
```
So, I think in regards to migration files, they don't want us editing already existing ones, or we risk breaking something. Also, any environment other than dev (so prod I think, not sure about others) will *not* run these, since they run migration files only once, at least per the docs in the "danger" box on this page: https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#understanding-migrations.

So all of these changes to mto_shipment_type would need to be in a new migration file, where you alter the type similar to this one. In fact you could probably just copy this script line for line and it would work, it just needs to be in a new migration file so that the updates work in production. Since these are all working on the same table you should just need a single migration file. 

Probably best to keep it separate from the create mobile home script since they're technically two distinct operations.

brooklyn-welsh commented on migrations/app/schema/20220203001924_add_ppm_to_mto_shipments_shipment_type_enum_values.up.sql:
```diff
@@ -7,7 +7,7 @@ CREATE TYPE mto_shipment_type_2 AS ENUM (
 	'HHG_INTO_NTS_DOMESTIC',
 	'HHG_OUTOF_NTS_DOMESTIC',
-	'MOTORHOME',
+	'MOBILEHOME',
```
See other comment regarding migrations.

brooklyn-welsh commented on migrations/app/schema/20230605163737_delete_hhg_shorthaul_domestic_hhg_longhaul_domestic_from_mto_shipments_shipment_type.up.sql:
```diff
@@ -10,7 +10,7 @@ CREATE TYPE mto_shipment_type_2 AS ENUM (
 	'HHG_INTO_NTS_DOMESTIC',
 	'HHG_OUTOF_NTS_DOMESTIC',
-	'MOTORHOME',
+	'MOBILEHOME',
```
See other comment regarding migrations.

taeJungCaci replied:
This might have been suggested already, but also might be a good idea so have it as `MOBILE_HOME` to be consistent with others.

brooklyn-welsh commented on swagger-def/definitions/MobileHome.yaml:
```diff
@@ -0,0 +1,68 @@
+  requestedDeliveryDate:
+    format: date-time
+    type: string
+  dimensions:
```
Since we already have fields for length, width, and height, is there any reason we need a string field for dimensions?

KonstanceH replied:
I just added it because it was in the agilty story details, but I agree it feels redundant so I'll remove it.

brooklyn-welsh commented on migrations/app/schema/20240723184614_create_mobile_home.up.sql:
```diff
@@ -0,0 +1,21 @@
+CREATE TABLE IF NOT EXISTS mobile_home (
```
I think we'll need created_at, updated_at, and deleted_at fields for this too, I know deleted_at is one that's used for soft deleting things, and I think updated_at is used for the E-Tag system they use for dealing with multiple people editing the same resource. Also recording these things might be a contractual obligation for records but I'm not sure.

brooklyn-welsh commented on swagger-def/ghc.yaml:
```diff
@@ -5975,11 +5975,68 @@ definitions:
+        type: number
+        format: double
+      width:
+        type: number
+        format: double
+      requestedPickupDate:
+        format: date
+        type: string
+        x-nullable: true
+      requestedDeliveryDate:
+        format: date
+        type: string
+        x-nullable: true
+      pickupAddress:
+        allOf:
+        - $ref: 'definitions/Address.yaml'
+      hasSecondaryPickupAddress:
+        type: boolean
+        x-nullable: true
+        x-omitempty: false
+      secondaryPickupAddress:
+        allOf:
+        - $ref: 'definitions/Address.yaml'
+      destinationAddress:
+        allOf:
+        - $ref: 'definitions/Address.yaml'
+      hasSecondaryDestinationAddress:
+        type: boolean
+        x-nullable: true
+        x-omitempty: false
+      secondaryDestinationAddress:
+        allOf:
+        - $ref: 'definitions/Address.yaml'
+      originAddress:
+        allOf:
+        - $ref: 'definitions/Address.yaml'
+      receivingAgent:
+        type: string
+      customerRemarks:
+        type: string
```
I think you could replace this block with "$ref: 'definitions/MobileHome.yaml".

brooklyn-welsh commented on swagger/ghc.yaml:
```diff
@@ -6226,11 +6226,68 @@ definitions:
+        type: number
+        format: double
+      width:
+        type: number
+        format: double
+      requestedPickupDate:
+        format: date
+        type: string
+        x-nullable: true
+      requestedDeliveryDate:
+        format: date
+        type: string
+        x-nullable: true
+      pickupAddress:
+        allOf:
+          - $ref: '#/definitions/Address'
+      hasSecondaryPickupAddress:
+        type: boolean
+        x-nullable: true
+        x-omitempty: false
+      secondaryPickupAddress:
+        allOf:
+          - $ref: '#/definitions/Address'
+      destinationAddress:
+        allOf:
+          - $ref: '#/definitions/Address'
+      hasSecondaryDestinationAddress:
+        type: boolean
+        x-nullable: true
+        x-omitempty: false
+      secondaryDestinationAddress:
+        allOf:
+          - $ref: '#/definitions/Address'
+      originAddress:
+        allOf:
+          - $ref: '#/definitions/Address'
+      receivingAgent:
+        type: string
+      customerRemarks:
+        type: string
```
Could replace with "$ref: '/definitions/MobileHome.yaml"

taeJungCaci reviewed (commented):
I haven't had the time to fully review this PR, but one of the things @brooklyn-welsh suggested was adding index to sql. This is what I had added on mine for Boat shipment:

CREATE INDEX boat_shipments_shipment_id_idx ON boat_shipments (shipment_id);
CREATE INDEX boat_shipments_deleted_at_idx ON boat_shipments (deleted_at);

Here is my PR for ref - I deleted other indexes not needed in my other PR:
https://github.com/transcom/mymove/pull/13296

taeJungCaci commented:
> I haven't had the time to fully review this PR, but one of the things @brooklyn-welsh suggested was adding index to sql. This is what I had added on mine for Boat shipment:
> 
> CREATE INDEX boat_shipments_shipment_id_idx ON boat_shipments (shipment_id); CREATE INDEX boat_shipments_deleted_at_idx ON boat_shipments (deleted_at);
> 
> Here is my PR for ref - I deleted other indexes not needed in my other PR: #13296

Also, just to share my approach, I am using only inches for my length, width, and height to make it easier and named the variables - `lengthInInches`, `widthInInches`, and `heightInInches`. Thought it would be more descriptive. If you had something easier or better, I may be able to take your approach.

KonstanceH commented:
> > I haven't had the time to fully review this PR, but one of the things @brooklyn-welsh suggested was adding index to sql. This is what I had added on mine for Boat shipment:
> > CREATE INDEX boat_shipments_shipment_id_idx ON boat_shipments (shipment_id); CREATE INDEX boat_shipments_deleted_at_idx ON boat_shipments (deleted_at);
> > Here is my PR for ref - I deleted other indexes not needed in my other PR: #13296
> 
> Also, just to share my approach, I am using only inches for my length, width, and height to make it easier and named the variables - `lengthInInches`, `widthInInches`, and `heightInInches`. Also in `*int` Thought it would be more descriptive. If you had something easier or better, I may be able to take your approach.

yea that's definitely more descriptive so I'll add them in.

KonstanceH commented:
> I haven't had the time to fully review this PR, but one of the things @brooklyn-welsh suggested was adding index to sql. This is what I had added on mine for Boat shipment:
> 
> CREATE INDEX boat_shipments_shipment_id_idx ON boat_shipments (shipment_id); CREATE INDEX boat_shipments_deleted_at_idx ON boat_shipments (deleted_at);
> 
> Here is my PR for ref - I deleted other indexes not needed in my other PR: #13296

ah okay I'll add this in too

brooklyn-welsh reviewed (approved):
LGTM!

taeJungCaci commented on migrations/app/schema/20240723184614_create_mobile_home.up.sql:
```diff
@@ -0,0 +1,31 @@
+	id uuid PRIMARY KEY NOT NULL,
+    shipment_id    uuid NOT NULL
+    CONSTRAINT mobile_home_mto_shipment_id_fkey
+    REFERENCES mto_shipments(id),
```
I think it would be good to add `ON DELETE CASCADE,` here so when the mtoShipment gets deleted, this will also get deleted.

```
CONSTRAINT mobile_home_mto_shipment_id_fkey
REFERENCES mto_shipments(id),
ON DELETE CASCADE,
```

taeJungCaci commented on migrations/app/schema/20240723184614_create_mobile_home.up.sql:
```diff
@@ -0,0 +1,31 @@
+COMMENT on COLUMN mobile_homes.deleted_at IS 'Date that the Mobile Home shipment was soft deleted.';
+
+CREATE INDEX mobile_homes_shipment_id_idx ON mobile_homes (shipment_id);
+CREATE INDEX mobile_homes_created_at_idx ON mobile_homes (created_at);
```
This one you could probably take out. Jim was recommending to remove the ones that may not be useful and could eat up space.

`CREATE INDEX mobile_homes_created_at_idx ON mobile_homes (created_at);`

taeJungCaci commented on pkg/models/mto_shipments.go:
```diff
@@ -34,8 +34,8 @@ const (
-	// MTOShipmentTypeMotorhome is a Shipment Type for Motorhome
-	MTOShipmentTypeMotorhome MTOShipmentType = "MOTORHOME"
+	// MTOShipmentTypeMobilehome is a Shipment Type for Mobilehome
+	MTOShipmentTypeMobilehome MTOShipmentType = "MOBILE_HOME"
```
Should this be `MTOShipmentTypeMobileHome` instead? I think you had Mobile_Home everywhere else. Might be better to change this now than later down the road where there might be 100+ files.

taeJungCaci commented on swagger-def/definitions/MobileHome.yaml:
```diff
@@ -0,0 +1,56 @@
+    format: date-time
+    type: string
+    readOnly: true
+  pickupAddress:
+    $ref: 'Address.yaml'
+  hasSecondaryPickupAddress:
+    type: boolean
+    x-omitempty: false
+    x-nullable: true
+  requestedPickupDate:
+    format: date-time
+    type: string
+  pickupLocation:
+    $ref: 'Address.yaml'
+  requestedDeliveryDate:
+    format: date-time
+    type: string
+  originAddress:
+    $ref: 'Address.yaml'
+  secondaryPickupAddress:
+    allOf:
+      - $ref: 'Address.yaml'
+      - x-nullable: true
+      - x-omitempty: false
```
I believe these are not available for mobile home and are only for PPMs

KonstanceH replied:
yea you're right I could've sworn I had already removed them, but I just pushed this change up

taeJungCaci reviewed (approved):
LGTM!
